### ISSUE TITLE
[BUG][Java Spring OAS3] Invalid allowable values generated for enum path variables

### ISSUE THREAD

**Issue by whydoievenneedthis** [OpenAPITools/openapi-generator#11163]
##### Description

Enum path variables with java spring oas3 option are generated incorrectly.

##### openapi-generator version

5.3.1

##### OpenAPI declaration file content

```yaml
openapi: 3.0.3
info:
  title: Api Documentation
  description: Api Documentation
  termsOfService: urn:tos
  contact: {}
  license:
    name: Apache 2.0
    url: http://www.apache.org/licenses/LICENSE-2.0
  version: '1.0'
paths:
  /thingy/{type}:
    get:
      operationId: get
      parameters:
        - name: type
          in: path
          description: type
          required: true
          schema:
            $ref: '#/components/schemas/Periodicity'
      responses:
        '200':
          description: OK
components:
  schemas:
    Periodicity:
      type: string
      enum:
        - MORNING
        - EVENING
        - DAILY
        - WEEKLY
        - BIWEEKLY
        - MONTHLY
```

##### Generation Details

Generator name: spring
Config options: oas3 = true

##### Generated java code

The following code fragment is generated with the oas3 option turned on:
```
@Parameter(
  name = "type",
  description = "type",
  required = true,
  allowableValues = "MORNING, EVENING, DAILY, WEEKLY, BIWEEKLY, MONTHLY"
)
```

##### Expected java code
```
@Parameter(
  name = "type",
  description = "type",
  required = true,
  schema = @Schema(allowableValues = "MORNING, EVENING, DAILY, WEEKLY, BIWEEKLY, MONTHLY")
)
```

**Comment by whydoievenneedthis** [comment]
Suggested pathParams.mustache file:

```
{{#isPathParam}}
    {{#useBeanValidation}}{{>beanValidationPathParams}}{{/useBeanValidation}}
    {{#oas3}}
        @Parameter(
            name = "{{baseName}}",
            {{#allowableValues}}schema = @Schema(allowableValues = "{{#enumVars}}{{#lambdaRemoveDoubleQuote}}{{{value}}}{{/lambdaRemoveDoubleQuote}}{{^-last}}, {{/-last}}{{#-last}}{{/-last}}{{/enumVars}}"),{{/allowableValues}}
            description =
    {{/oas3}}
    {{^oas3}}
        @ApiParam(
            {{#allowableValues}}allowableValues = "{{#enumVars}}{{#lambdaRemoveDoubleQuote}}{{{value}}}{{/lambdaRemoveDoubleQuote}}{{^-last}}, {{/-last}}{{#-last}}{{/-last}}{{/enumVars}}",{{/allowableValues}}
            value =
    {{/oas3}}

        "{{{description}}}"
        {{#required}},required = true{{/required}}

        {{^isContainer}}
            {{#defaultValue}}, defaultValue = "{{{.}}}"{{/defaultValue}}
        {{/isContainer}}
    )
    @PathVariable("{{baseName}}")
    {{#isDate}} @org.springframework.format.annotation.DateTimeFormat(iso = org.springframework.format.annotation.DateTimeFormat.ISO.DATE){{/isDate}}
    {{>optionalDataType}} {{paramName}}
{{/isPathParam}}
```

**Comment by wing328** [comment]
Closed via #11181. Please pull the latest master to give it another try.
